### PR TITLE
[WIP] Format code in risky rule reasons

### PR DESCRIPTION
**Pull request by ntzm**
In RST, you can't have code in italics blocks, so we have a lot of `` in the readme. I propose we only italic the `Risky rule:` part of it, so the rest can be formatted correctly.

### CODE REVIEW

**Comment by ntzm** [comment]
Alternatively, we could replace 
\*Risky rule: foo \`\`bar\`\` baz\*
with
\*Risky rule: foo\* \`\`bar\`\` \*baz\*
which would keep the italics throughout, just terminating before each code block and starting after

**Comment by ntzm** [comment]
My regex skills are failing me 😢 I've got `$help = Preg::replace('#(\`\`[^`]+\`\`) #m', '* $1 *', $help);` to terminate the `*` before and start it up again, but I need to use a lookbehind to assert the line starts with a `*`. Any help?

File: README.rst [hunk 1]
```diff
@@ -1643,25 +1643,25 @@ Choose from the list of available rules:
 
   Lambdas not (indirect) referencing ``$this`` must be declared ``static``.
 
-  *Risky rule: risky when using "->bindTo" on lambdas without referencing to ``$this``.*
+  *Risky rule:* risky when using "->bindTo" on lambdas without referencing to ``$this``.
```
Comment (dmvdbrugge) [inline README.rst]: Unrelated, but shouldn't this be ` ``->bindTo`` `?

Comment (ntzm) [reply README.rst]: Yes it should 😁

**Comment by dmvdbrugge** [comment]
Lookbehind is relative to the thing you're trying to match, which would make looking back to the start of the line "non-fixed width", which is an error, according to https://regex101.com/r/Ji8pRi/1/

Also that space in your regex is wrong, as can be seen in the README.rst part commented on above, it would fail the ` ``this``.`

I don't know if this can be solved with regexes, but then again I'm not that good with 'em (while this codebase is full of them 😱).

**Comment by SpacePossum** [comment]
a more stable solution could be to create a new style tag `<risky>`,
add the style to the `HelpCommand` ; `->setStyle('risky', new OutputFormatterStyle('white', 'red'));`
update `"   | *Risky rule: %s.*\n",` to `"   | <risky>Risky rule</risky>: %s.\n",`
now in the `ReadmeCommand` you can "just" match on the tags, making your life more easy ;)

**Comment by kubawerlos** [comment]
Does this PR still make any sense since we have https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3475? I think we can close it.

**Comment by SpacePossum** [comment]
@ntzm closing for now as there hasn't been worked on this for a bit and we have another PR open for changes on this topic, let me know if you plan on working it though than we can reopen the PR, thanks for all the work!

**Pull request closed without merging**